### PR TITLE
refactor: express api app

### PR DESCRIPTION
**Pull request by 0xnigir1**
## Description

Refactor API app into an Express server

- Drops NestJS structure
- Leaves functionality as it is but using an Express server
- Setup Middlewares and OpenApi
- Use Zod for Schema and Env validation
- Use `tsx` for hot-reload server (dev mode)

### CODE REVIEW

File: apps/api/src/index.ts [hunk 1]
```diff
@@ -1,16 +1,28 @@
...
-        return res.json({ message: `/metrics/ecosystem/${req.params.chainId}` });
-    });
+import { App } from "./app.js";
+import { config } from "./common/config/index.js";
+import { Container } from "./container.js";
 
-server.listen(port, () => {
-    console.log(`api running on ${port}`);
+const container = new Container(config);
+const logger = container.get<ILogger>("ILogger");
```
Comment (0xkenj1) [inline apps/api/src/index.ts:10]: wdyt if we remove this class and do all the instantiations on the `main` function, i think it is more readable and better to understand

File: apps/api/src/metrics/routes/index.ts [hunk 2]
```diff
@@ -0,0 +1,335 @@
...
+//                     message: error.message,
+//                 });
+//             }
+//             next(error);
+//         }
+//     });
+
+//     return router;
+// };
+
```
Comment (0xkenj1) [inline apps/api/src/metrics/routes/index.ts:45]: are this comments intentional ?

File: apps/api/src/metrics/routes/index.ts
Comment (0xkenj1) [inline apps/api/src/metrics/routes/index.ts:1]: swagger definitions in the code introduces tons of noise, is there a way to have the definitions in `openApi/metrics.yaml` for example ?

File: apps/api/src/metrics/routes/index.ts (hunk 2 above)
Comment (0xnigir1) [reply apps/api/src/metrics/routes/index.ts:45]: my bad though i've deleted it

File: apps/api/src/metrics/routes/index.ts
Comment (0xnigir1) [reply apps/api/src/metrics/routes/index.ts:1]: i'll look into it 🫡

File: apps/api/src/index.ts (hunk 1 above)
Comment (0xyaco) [reply apps/api/src/index.ts:10]: I'd say that I agree.

Do you envision the `Container` class having another responsibilities apart from being used during the app initialization?

**Review by 0xyaco — APPROVED** [review]

**Review by 0xkenj1 — APPROVED** [review]

**Pull request merged**